lib/stringify.js

```diff
@@ -220,16 +233,27 @@ module.exports = function (object, opts) {
         return '';
     }
 
-    var arrayFormat;
-    if (opts && opts.arrayFormat in arrayPrefixGenerators) {
-        arrayFormat = opts.arrayFormat;
+    var generateArrayPrefix;
+    if (opts && typeof opts.arrayFormat === 'function') {
+        generateArrayPrefix = opts.arrayFormat;
+    } else if (opts && opts.arrayFormat in arrayPrefixGenerators) {
+        generateArrayPrefix = arrayPrefixGenerators[opts.arrayFormat];
     } else if (opts && 'indices' in opts) {
-        arrayFormat = opts.indices ? 'indices' : 'repeat';
+        generateArrayPrefix = arrayPrefixGenerators[opts.indices ? 'indices' : 'repeat'];
     } else {
-        arrayFormat = 'indices';
+        generateArrayPrefix = arrayPrefixGenerators.indices;
     }
 
-    var generateArrayPrefix = arrayPrefixGenerators[arrayFormat];
+    var generateObjectPrefix;
+    if (opts && typeof opts.objectFormat === 'function') {
+        generateObjectPrefix = opts.objectFormat;
+    } else if (opts && opts.objectFormat in objectPrefixGenerators) {
+        generateObjectPrefix = objectPrefixGenerators[opts.objectFormat];
+    } else if (opts && opts.allowDots) {
+        generateObjectPrefix = objectPrefixGenerators.dots;
+    } else {
+        generateObjectPrefix = objectPrefixGenerators.brackets;
+    }
```

Comment:
this seems like it should be handled by the option normalization and the default?